core/meta/src/TClass.cxx

```diff
@@ -3102,6 +3102,17 @@ TClass *TClass::GetClass(const char *name, Bool_t load, Bool_t silent, size_t hi
       loadedcl = LoadClassDefault(normalizedName.c_str(),silent);
    } else {
       if (gInterpreter->AutoLoad(normalizedName.c_str(),kTRUE)) {
+         // At this point more infomration has been loaded.  This
+         // information might be pertinent ot the normalization of the name.
+         // For example it might contain or be a typedef for which we don't
+         // have a forward declaration (eg. typedef to instance of class
+         // template with default paramters).  So let's redo the normalization
+         // as the new information (eg typedef in TROOT::GetListOfTypes) might
+         // lead to a different value.
+         {
+            TInterpreter::SuspendAutoLoadingRAII autoloadOff(gInterpreter);
+            TClassEdit::GetNormalizedName(normalizedName, name);
+         }
```

Comment:
IIUC, this tries to normalize the type name again (first try in https://github.com/root-project/root/blob/47e277a3fb88e8845eebf90bca207f0c3bed904d/core/meta/src/TClass.cxx#L3053-L3056), in case new information became available.

However, what if we have a `typedef` that aliases a `typedef` (that aliases ...), that names a class.  Would that require additional steps?

Reply:
Also, unrelated to the additions in the patch: `TClass:GetClass()` code seems already hard to follow (incl. special handling for some cases).  This adds to it.
I think it should be refactored at some point (?)

Reply:
> IIUC, this tries to normalize the type name again

Yes, that the main point of this PR.

> However, what if we have a typedef that aliases a typedef (that aliases ...), that names a class. Would that require additional steps?

It might but not.  Presumably all the dependent library with loaded during the "explicit" autoload right above.  And no new libraries would be loaded as part of this line because of the suspension.